src/app/pages/Languages.svelte

```diff
@@ -51,24 +58,60 @@
     selectedButton = buttonName;
   };
 
+  /**
+   * Creates an anchor to download a file programmatically
+   * @param href link to Blob or File
+   * @param download Title of the download
+   *
+   */
+  const createAnchor = (href: string, download: string): void => {
+    const anchor = document.createElement("a");
+    anchor.href = href;
+    anchor.target = "_blank";
+    anchor.download = download;
+    // Auto click on a element, trigger the file download
+    anchor.click();
+    anchor.remove();
+  };
```

Comment:
Why use this instead of having this in the markup?

```jsx
<a class="button" download={`${dictionaryName}.kmp`} href={$currentDownloadURL}>
```

Reply:
Yes that's true, changing now

Reply:
The thing with doing the markup is that dictionaryName will need to have a listener to insure it is up to date. This means sending the info up from LanguageInfo or keeping it in the store. Alternatively doing the anchor programmatically allows us to just retrieve the name when needed.  The createAnchor logic is already used by handleExport too.

Reply:
Perhaps a better solution is to refactor: **extract a component** called `<Download href={...}></Download>` that is contains the `createAnchor` logic, but _not_ the logic to create a download name. I feel that creating DOM elements that are never mounted to the document just to use its side-effects is hacky, and may inadvertantly lead to accessibility issues.